Typescript 4.8.2 does not like the type definitions from sequelize 6.21.4

Upgrading to Typescripte 4.8.2 and running tsc prints on my system:
```
node_modules/sequelize/types/hooks.d.ts:69:19 - error TS2344: Type 'TAttributes' does not satisfy the constraint '{}'.                  
                                                                                                                                        
69   M extends Model<TAttributes, TCreationAttributes> = Model,                                                                         
                     ~~~~~~~~~~~                                                                                                        
                                                                                                                                        
  node_modules/sequelize/types/hooks.d.ts:70:3                                                                                          
    70   TAttributes = any,                                                                                                             
         ~~~~~~~~~~~~~~~~~                                                                                                              
    This type parameter might need an `extends {}` constraint.                                                                          
                                                                                                                                        
node_modules/sequelize/types/model.d.ts:3252:105 - error TS2344: Type 'TModelAttributes' does not satisfy the constraint '{}'.          
                                                                                                                                        
3252 export type ModelType<TModelAttributes = any, TCreationAttributes = TModelAttributes> = new () => Model<TModelAttributes, TCreation
Attributes>;
                                                                                                             ~~~~~~~~~~~~~~~~

  node_modules/sequelize/types/model.d.ts:3252:23
    3252 export type ModelType<TModelAttributes = any, TCreationAttributes = TModelAttributes> = new () => Model<TModelAttributes, TCrea
tionAttributes>;
                               ~~~~~~~~~~~~~~~~~~~~~~
    This type parameter might need an `extends {}` constraint.

node_modules/sequelize/types/model.d.ts:3260:52 - error TS2344: Type 'S' does not satisfy the constraint '{}'.

3260 export type ModelDefined<S, T> = ModelStatic<Model<S, T>>;
                                                        ~

  node_modules/sequelize/types/model.d.ts:3260:26
    3260 export type ModelDefined<S, T> = ModelStatic<Model<S, T>>;
                                  ~
    This type parameter might need an `extends {}` constraint.
```
As these errors do not reference my files, I assume that the files with type definitions exported by sequelize 6.21.4 are not good.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -111,7 +111,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        ts-version: ["4.4", "4.5", "4.6", "4.7"]
+        ts-version: ["4.5", "4.6", "4.7", "4.8"]
     name: TS Typings (${{ matrix.ts-version }})
     runs-on: ubuntu-latest
     needs: lint

--- a/package.json
+++ b/package.json
@@ -115,7 +115,7 @@
     "tedious": "15.1.0",
     "ts-node": "10.9.1",
     "typedoc": "0.23.14",
-    "typescript": "4.7.4"
+    "typescript": "4.8.3"
   },
   "peerDependenciesMeta": {
     "pg": {

--- a/src/hooks.d.ts
+++ b/src/hooks.d.ts
@@ -74,8 +74,8 @@ export interface ModelHooks<M extends Model = Model, TAttributes = any> {
 
 export interface SequelizeHooks<
   M extends Model<TAttributes, TCreationAttributes> = Model,
-  TAttributes = any,
-  TCreationAttributes = TAttributes,
+  TAttributes extends {} = any,
+  TCreationAttributes extends {} = TAttributes,
 > extends ModelHooks<M, TAttributes> {
   beforeDefine(attributes: ModelAttributes<M, TCreationAttributes>, options: ModelOptions<M>): void;
   afterDefine(model: ModelStatic): void;

--- a/src/model.d.ts
+++ b/src/model.d.ts
@@ -3592,7 +3592,7 @@ export abstract class Model<TModelAttributes extends {} = any, TCreationAttribut
   isSoftDeleted(): boolean;
 }
 
-export type ModelDefined<S, T> = ModelStatic<Model<S, T>>;
+export type ModelDefined<S extends {}, T extends {}> = ModelStatic<Model<S, T>>;
 
 // remove the existing constructor that tries to return `Model<{},{}>` which would be incompatible with models that have typing defined & replace with proper constructor.
 export type ModelStatic<M extends Model = Model> = OmitConstructors<typeof Model> & { new(): M };

--- a/src/utils/object.ts
+++ b/src/utils/object.ts
@@ -125,7 +125,7 @@ export function cloneDeep<T extends object>(obj: T, onlyPlain?: boolean): T {
  * @returns a flattened object
  * @private
  */
-export function flattenObjectDeep<T>(value: T): T extends object ? Flatten<T> : T {
+export function flattenObjectDeep<T extends {}>(value: T): T extends object ? Flatten<T> : T {
   if (!isPlainObject(value)) {
     // TypeScript doesn't know T is an object due to isPlainObject's typings. Cast to any.
     return value as any;
@@ -218,6 +218,6 @@ export function camelizeObjectKeys(obj: { [key: string]: any }) {
 
 type NoUndefinedField<T> = { [P in keyof T]: Exclude<T[P], null | undefined> };
 
-export function removeUndefined<T>(val: T): NoUndefinedField<T> {
+export function removeUndefined<T extends {}>(val: T): NoUndefinedField<T> {
   return omitBy(val, isUndefined) as NoUndefinedField<T>;
 }

--- a/src/utils/sql.ts
+++ b/src/utils/sql.ts
@@ -242,6 +242,7 @@ function mapBindParametersAndReplacements(
         throw new Error(`Positional replacement (?) ${replacementIndex} has no entry in the replacement map (replacements[${replacementIndex}] is undefined).`);
       }
 
+      // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
       const escapedReplacement = escapeSqlValue(replacementValue as any, undefined, dialect.name, true);
 
       // add everything before the bind parameter name

--- a/test/support.ts
+++ b/test/support.ts
@@ -58,7 +58,7 @@ function inlineErrorCause(error: Error) {
   // eslint-disable-next-line @typescript-eslint/prefer-ts-expect-error
   // @ts-ignore -- TS < 4.6 doesn't include the typings for this property, but TS 4.6+ does.
   const cause = error.cause;
-  if (cause) {
+  if (cause instanceof Error) {
     message += `\nCaused by: ${inlineErrorCause(cause)}`;
   }
 

